tests/cli/bootsys/test_mgmt_power.py

```diff
@@ -177,3 +183,160 @@ def test_mgmt_ncns_skip_prompt_power_off(self):
+        self.assertEqual(cm.records[2].message, 'Shutting down manager NCNs: ncn-m002, ncn-m003')
+        self.assertEqual(cm.records[3].message, 'Waiting up to 1200 seconds for manager NCNs to shut down...')
+        self.assertEqual(cm.records[4].message, 'Freezing Ceph and shutting down storage NCNs: ncn-s001, ncn-s002')
+        self.assertEqual(cm.records[5].message, 'Ceph freeze completed successfully on storage NCNs.')
+        self.assertEqual(cm.records[6].message, 'Waiting up to 1200 seconds for storage NCNs to shut down...')
+        self.assertEqual(cm.records[7].message, 'Shutdown and power off of storage NCNs: ncn-s001, ncn-s002')
+        self.assertEqual(cm.records[8].message, 'Shutdown and power off of all management NCNs complete.')
+
+    def test_do_mgmt_shutdown_power_with_fatal_error(self):
+        """Test do_mgmt_shutdown_power when a fatal error occurs."""
+        self.mock_do_ceph_freeze.side_effect = FatalPlatformError('Ceph freeze failed')
+        with self.assertLogs(level=logging.ERROR) as cm:
+            with self.assertRaises(SystemExit):
+                do_mgmt_shutdown_power(self.username, self.password, self.mock_args.excluded_ncns,
+                                       self.ncn_shutdown_timeout, self.ipmi_timeout)
+
+        self.mock_get_and_verify_ncn_groups.assert_called_once_with(self.mock_args.excluded_ncns.union({'ncn-m001'}))
+        self.assertEqual(cm.records[-1].message, 'Failed to freeze Ceph on storage NCNs: Ceph freeze failed')
+
+    def test_set_next_boot_device_to_disk_success(self):
+        """Test that the function sets the boot device to disk successfully"""
+        mock_ssh_client = MagicMock()
+        mock_stdout = MagicMock()
+        mock_stderr = MagicMock()
+        mock_stdout.channel.recv_exit_status.return_value = 0
+        mock_stdout.read.return_value = b"Boot0001* UEFI OS\nBoot0002* Other OS"
+
+        mock_ssh_client.exec_command.return_value = (None, mock_stdout, mock_stderr)
+
+        ncns = ['ncn-w001', 'ncn-s001']
+        set_next_boot_device_to_disk(mock_ssh_client, ncns)
+
+        expected_connect_calls = [call('ncn-w001'), call('ncn-s001')]
+        mock_ssh_client.connect.assert_has_calls(expected_connect_calls, any_order=True)
+
+        expected_exec_calls = [
+            call('efibootmgr'),
+            call('efibootmgr -n 0001'),
+            call('efibootmgr'),
+            call('efibootmgr -n 0001')
+        ]
+        mock_ssh_client.exec_command.assert_has_calls(expected_exec_calls, any_order=True)
+
+    def test_set_next_boot_device_to_disk_ssh_fail(self):
+        """Test that the function handles SSH connection failures"""
+        mock_ssh_client = MagicMock()
+        mock_ssh_client.connect.side_effect = SSHException('ssh failed')
+
+        with patch('sat.cli.bootsys.mgmt_power.LOGGER') as mock_logger:
+            ncns = ['ncn-w001']
```

Comment:
You may want to expand this test to call `set_next_boot_device_to_disk` with more than one NCN hostname to show that it will continue with the other NCNs when one fails.